### PR TITLE
Added tests for GenerateDirectoryTree function in the scanner package.

### PR DESCRIPTION
**Pull request by CynthiaFotso**
## Description:
This PR adds tests to the scanner package file for the `GenerateDirectoryTree` which was not tested before.

## Changes Made:
Added tests for the scenarios below: 
- Empty files
- Single file
- Single directory
- File with tokens displayed
- File with zero tokens
- Mixed files and directories
- Special characters in file/directory names
- Paths sorted alphabetically

## Testing:

- The new tests pass locally with `go test -v ./pkg/scanner/`
- No existing tests were modified or broken.
- Verified with `go test -cover ./pkg/scanner/` to ensure coverage remains adequate and the coverage improved from 8.2% to 30.3%.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests for directory tree generation, covering edge cases including empty input, single files/directories, token counting, special characters, and alphabetical sorting. Tests validate output formatting and token display logic.

### CODE REVIEW

File: pkg/scanner/scanner_test.go [hunk 1]
```diff
@@ -304,3 +304,162 @@ func TestBuildPathMap_DuplicatePaths(t *testing.T) {
...
+
+	// When
+	result := generateDirectoryTree(files, rootPath)
+
+	// Then
+	expected := "a.txt\nm/\nz.txt\n"
+	if result != expected {
+		t.Errorf("Expected %q, got %q", expected, result)
+	}
+}
```
[the remark this answers is not in the document]
Comment (BHChen24) [reply pkg/scanner/scanner_test.go:465]: enough good for the test now, will add more tests for nested paths later
